src/olmo_core/train/callbacks/evaluator_callback.py

```diff
@@ -119,16 +126,18 @@ def _perform_eval(self):
                     with cuda_sync_debug_mode(0):
                         evaluator.update_metrics(batch, ce_loss, logits)
 
-                if eval_step % self.trainer.cancel_check_interval == 0:
-                    self.trainer.check_if_canceled()
+                if (
+                    not self.cancel_after_first_eval
+                ):  # guard here in case we have multiple evaluators doing this
+                    if self.trainer.is_canceled:
+                        self._log_progress(evaluator, eval_step)
+                        return
```

Comment:
I find this confusing. If multiple callbacks cancel the trainer, we still want to run them all, right? The canceling only takes effect when it's time to do the next step?

Reply:
I just removed this block: a8de0efa

We really don't need it. I had it there initially because I thought a run could get canceled during an eval loop, in which case we should break out of the loop early. But now I think it's better to finish the eval loop.